payload/builder/payload.go

```diff
@@ -27,26 +27,34 @@ import (
 
 	ctypes "github.com/berachain/beacon-kit/consensus-types/types"
 	engineprimitives "github.com/berachain/beacon-kit/engine-primitives/engine-primitives"
+	"github.com/berachain/beacon-kit/payload/attributes"
 	"github.com/berachain/beacon-kit/primitives/common"
 	"github.com/berachain/beacon-kit/primitives/math"
-	statedb "github.com/berachain/beacon-kit/state-transition/core/state"
 )
 
 // RequestPayloadAsync builds a payload for the given slot and
 // returns the payload ID.
 func (pb *PayloadBuilder) RequestPayloadAsync(
 	ctx context.Context,
-	st *statedb.StateDB,
-	slot math.Slot,
+	st attributes.ReadOnlyBeaconState,
```

Comment:
imo this package should define ReadOnlyBeaconState by itself and not use the interface defined in a sibling package

Reply:
Agree, cleaned up interfaces